ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/IncrementalLoad.java

```diff
@@ -0,0 +1,307 @@
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.ql.DriverContext;
+import org.apache.hadoop.hive.ql.exec.Task;
+import org.apache.hadoop.hive.ql.exec.TaskFactory;
+import org.apache.hadoop.hive.ql.exec.repl.ReplStateLogWork;
+import org.apache.hadoop.hive.ql.exec.repl.bootstrap.events.filesystem.IncrementalLoadEventsIterator;
+import org.apache.hadoop.hive.ql.hooks.ReadEntity;
+import org.apache.hadoop.hive.ql.hooks.WriteEntity;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.ql.parse.ReplicationSpec;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+import org.apache.hadoop.hive.ql.parse.repl.DumpType;
+import org.apache.hadoop.hive.ql.parse.repl.ReplLogger;
+import org.apache.hadoop.hive.ql.parse.repl.load.DumpMetaData;
+import org.apache.hadoop.hive.ql.parse.repl.load.UpdatedMetaDataTracker;
+import org.apache.hadoop.hive.ql.parse.repl.load.log.IncrementalLoadLogger;
+import org.apache.hadoop.hive.ql.parse.repl.load.message.MessageHandler;
+import org.apache.hadoop.hive.ql.plan.AlterDatabaseDesc;
+import org.apache.hadoop.hive.ql.plan.AlterTableDesc;
+import org.apache.hadoop.hive.ql.plan.DDLWork;
+import org.apache.hadoop.hive.ql.plan.DependencyCollectionWork;
+import org.apache.hadoop.hive.ql.stats.StatsUtils;
+import org.slf4j.Logger;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.HashSet;
+
+/**
+ * IncrementalLoad
+ * Iterate through the dump directory and create tasks to load the events.
+ */
+public class IncrementalLoad {
+  private final String dbName, tableName;
+  private final IncrementalLoadEventsIterator iterator;
+  private HashSet<ReadEntity> inputs;
+  private HashSet<WriteEntity> outputs;
+  private Logger log;
+  private final HiveConf conf;
+  private final String loadPath;
+
+  public IncrementalLoad(String dbName, String tableName, String loadPath,
+                         IncrementalLoadEventsIterator iterator, HiveConf conf) {
+    this.dbName = dbName;
+    this.tableName = tableName;
+    this.iterator = iterator;
+    inputs = new HashSet<>();
+    outputs = new HashSet<>();
+    log = null;
+    this.conf = conf;
+    this.loadPath = loadPath;
+  }
+
+  public Task<? extends Serializable> execute(DriverContext driverContext, Hive hive, Logger log) throws Exception {
+    int maxTasks = conf.getIntVar(HiveConf.ConfVars.REPL_APPROX_MAX_LOAD_TASKS);
+    Task<? extends Serializable> evTaskRoot = TaskFactory.get(new DependencyCollectionWork());
+    Task<? extends Serializable> taskChainTail = evTaskRoot;
+    Long lastReplayedEvent = null;
+    this.log = log;
+
+    ReplLogger replLogger = new IncrementalLoadLogger(dbName, loadPath, iterator.getNumEvents());
```

Comment:
replLogger should be persisted in the class private variable as it should live across iterations.

Reply:
done